cmd/ping-community/main.go

```diff
@@ -0,0 +1,459 @@
+	if err != nil {
+		logger.Error("failed to update mailservers", "error", err)
+		return
+
+	}
+	messenger.SetMailserver(mailserverNode.ID().Bytes())
+
+	time.Sleep(5 * time.Second)
+	community, err := messenger.RequestCommunityInfoFromMailserver(*communityID)
+	if err != nil {
+
+		logger.Error("community error", "error", err)
+		return
+
+	}
+	chat := community.Chats()[*chatID]
+	if chat == nil {
+		logger.Warn("Chat not found")
+		return
+	}
+	logger.Info("GOT community", "comm", chat)
+
+	response, err := messenger.JoinCommunity(context.Background(), community.ID())
+	if err != nil {
+		logger.Error("failed to join community", "err", err)
+	}
+
+	var targetChat *protocol.Chat
+
+	for _, c := range response.Chats() {
+		if strings.Contains(c.ID, *chatID) {
+			targetChat = c
+		}
+	}
+
+	if targetChat == nil {
+		logger.Warn("chat not found")
+		return
+	}
+
+	id := uuid.New().String()
+
+	ticker := time.NewTicker(2 * time.Second)
+	count := 0
+
+	for { // nolint: gosimple
+		select {
+		case <-ticker.C:
+			count++
+			timestamp := time.Now().Format(time.RFC3339)
+			logger.Info("Publishing", "id", id, "count", count, "time", timestamp)
+			inputMessage := &common.Message{}
+
+			inputMessage.Text = fmt.Sprintf("%d\n%s\n%s", count, timestamp, id)
+			inputMessage.LocalChatID = targetChat.ID
+			inputMessage.ChatId = targetChat.ID
+			inputMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+
+			_, err := messenger.SendChatMessage(context.Background(), inputMessage)
+			if err != nil {
+				logger.Error("failed to send a message", "err", err)
+			}
+
+		}
+	}
+
+}
+
+func getDefaultDataDir() string {
+	if home := os.Getenv("HOME"); home != "" {
+		return filepath.Join(home, ".statusd")
+	}
+	return "./statusd-data"
+}
+
+func setupLogging(config *params.NodeConfig) {
+	if *logLevel != "" {
+		config.LogLevel = *logLevel
+	}
+
+	logSettings := logutils.LogSettings{
+		Enabled:         config.LogEnabled,
+		MobileSystem:    config.LogMobileSystem,
+		Level:           config.LogLevel,
+		File:            config.LogFile,
+		MaxSize:         config.LogMaxSize,
+		MaxBackups:      config.LogMaxBackups,
+		CompressRotated: config.LogCompressRotated,
+	}
+	colors := !(*logWithoutColors) && terminal.IsTerminal(int(os.Stdin.Fd()))
+	if err := logutils.OverrideRootLogWithConfig(logSettings, colors); err != nil {
+		stdlog.Fatalf("Error initializing logger: %v", err)
+	}
+}
+
+// printVersion prints verbose output about version and config.
+func printVersion(config *params.NodeConfig) {
+	fmt.Println(strings.Title(config.Name))
+	fmt.Println("Version:", config.Version)
+	fmt.Println("Network ID:", config.NetworkID)
+	fmt.Println("Go Version:", runtime.Version())
+	fmt.Println("OS:", runtime.GOOS)
+	fmt.Printf("GOPATH=%s\n", os.Getenv("GOPATH"))
+	fmt.Printf("GOROOT=%s\n", runtime.GOROOT())
+
+	fmt.Println("Loaded Config: ", config)
+}
+
+func printUsage() {
+	usage := `
+Usage: statusd [options]
```

Comment:
This is just copy-pasted from other commands. Wouldn't it be good to update this and describe what this actually does?